project/src/main/kotlin/org/ionproject/core/calendar/CalendarData.kt

```diff
@@ -0,0 +1,286 @@
+        $DTSTAMP,
+        $CREATED,
+        $DTSTART,
+        $DTEND,
+        $DUE,
+        $BYDAY
+    """
+
+    private const val GROUP_BY = """GROUP BY
+        $UID, 
+        $TYPE,
+        $DTSTAMP,
+        $CREATED,
+        $DTSTART,
+        $DTEND,
+        $DUE,
+        $BYDAY"""
+
+    private const val CLASS_ABBREVIATION = "c"
+    private const val CLASS = "dbo.Class $CLASS_ABBREVIATION"
+    private const val CLASS_CALENDAR_ID = "$CLASS_ABBREVIATION.calendar"
+    private const val CLASS_COURSE = "$CLASS_ABBREVIATION.courseId"
+    private const val CLASS_TERM = "$CLASS_ABBREVIATION.term"
+
+    private const val CLASS_SECTION_ABBREVIATION = "cs"
+    private const val CLASS_SECTION = "dbo.Class $CLASS_SECTION_ABBREVIATION"
+    private const val CLASS_SECTION_CALENDAR_ID = "$CLASS_SECTION_ABBREVIATION.calendar"
+    private const val CLASS_SECTION_ID = "$CLASS_SECTION_ABBREVIATION.id"
+    private const val CLASS_SECTION_COURSE = "$CLASS_SECTION_ABBREVIATION.courseId"
+    private const val CLASS_SECTION_TERM = "$CLASS_SECTION_ABBREVIATION.term"
+
+    const val CALENDAR_FROM_CLASS_QUERY = """WITH calendar_id AS (
+                    SELECT $CLASS_CALENDAR_ID
+                    FROM $CLASS
+                    WHERE $CLASS_COURSE = :courseId AND $CLASS_TERM = :term
+                )
+            $SELECT
+            FROM
+                $CALENDAR_COMPONENT
+            WHERE $CALENDARS = (SELECT * FROM calendar_id)
+            $GROUP_BY"""
+
+    const val CALENDAR_COMPONENT_FROM_CLASS_QUERY = """WITH calendar_id AS (
+                    SELECT $CLASS_CALENDAR_ID
+                    FROM $CLASS
+                    WHERE $CLASS_COURSE = :courseId AND $CLASS_TERM = :term
+                )
+            $SELECT
+            FROM $CALENDAR_COMPONENT
+            $GROUP_BY
+            WHERE $CALENDARS = (SELECT * FROM calendar_id) AND $UID = :componentId"""
+
+    const val CALENDAR_FROM_CLASS_SECTION_QUERY =
+        """WITH calendar_id AS (
+                    SELECT $CLASS_SECTION_CALENDAR_ID
+                    FROM $CLASS_SECTION
+                    WHERE $CLASS_SECTION_COURSE = :courseId AND $CLASS_SECTION_TERM = :term AND $CLASS_SECTION_ID = :classSectionId
+                )
+            $SELECT
+            FROM $CALENDAR_COMPONENT
+            $GROUP_BY
+            WHERE $CALENDARS = (SELECT * FROM calendar_id)"""
+
+    const val CALENDAR_COMPONENT_FROM_CLASS_SECTION_QUERY = """WITH calendar_id AS (
+                    SELECT $CLASS_SECTION_CALENDAR_ID
+                    FROM $CLASS_SECTION
+                    WHERE $CLASS_SECTION_COURSE = :courseId AND $CLASS_SECTION_TERM = :term AND $CLASS_SECTION_ID = :classSectionId
+                )
+            $SELECT
+            FROM $CALENDAR_COMPONENT
+            $GROUP_BY
+            WHERE $CALENDARS = (SELECT * FROM calendar_id) AND $UID = :componentId"""
+
+    @Component
+    class CalendarComponentMapper(
```

Comment:
This class is already extense as it is, I recommend to not mix the Mapper with it and move it somewhere else